packages/automation-kit/README.md

```diff
@@ -2,10 +2,158 @@
 
 This library was generated with [Nx](https://nx.dev).
 
+
 ## Building
 
 Run `nx build automation-kit` to build the library.
 
+
 ## Running unit tests
 
 Run `nx test automation-kit` to execute the unit tests via [Vitest](https://vitest.dev/).
+
+---
+
+
+## Automation-kit Architecture
+
+The `automation-kit` provides a set of high-level functions to interact with smart contracts, designed for seamless creation and management of Ditto automation workflows.
+
+It abstracts blockchain interactions like transaction building, signing, and sending via a single adapter (e.g., ethers.js, web3.js, or viem) under the hood, ensuring that all smart contract calls consistently use the same adapter.
+
+---
+
+### `automation-kit` API
+
+#### SmartWalletFactory (`VaultFactoryABI.json`)
+
+- **`predictVaultAddress`**
+- **`deploy`**
+- **`getVersionAndIdByAddress`**
+
+
+#### SmartWallet (`VaultABI.json`)
+
+- **`deposit`**
+- **`withdraw`**
+- **`upgrade`**
+- **`executeOperation`**
+
+
+#### WorkflowFactory (`VaultABI.json`)
+
+- **`createWorkflow (buildAndDeploy)`** -> addWorkflowAndGelatoTask
+
+---
+
+
+### Example Code of Automation-Kit
+
+```typescript
+import { contractCall } from './web3-adapter';
+import { VaultFactoryABI } from './abis/VaultFactoryABI.json';
+
+// Predict the vault address
+async function predictVaultAddress(creator: string, vaultId: number) {
+  const result = await contractCall({
+    address: '0xVaultFactoryAddress',
+    abi: VaultFactoryABI,
+    method: 'predictDeterministicVaultAddress',
+    args: [creator, vaultId],
+    type: 'view',
+  });
+  return result;
+}
+```
+
+
+### Example of Automation-Kit Usage
+
+1. **Initialize Automation-Kit**: Users initialize the Automation Kit through a setup `createAutomationKit` method. This method will allow users to choose a library (e.g., ethers.js or web3.js).
+
+```typescript
+import { createAutomationKit, MultiSenderAction } from '@ditto-network/automation-kit';    // Core action
+import { EthersAdapter } from '@ditto-network/automation-kit-ethers';
+import { UniswapSwapAction } from 'ditto-automation-uniswap-action';  // External action
+
+
+// Initialize Automation-kit
+const automationKit = createAutomationKit({
+  adapter: new EthersAdapter(),
+  // register actions
+  actions: [MultiSenderAction, UniswapSwapAction],
```

Comment:
We don't need to register actions or triggers
We just call them when creating workflow